panther-labs/panther_analysis_tool
# CreateAlert support

### Background

Rules can now have a CreateAlert field that defaults to True. When set to False, it will suppress the alert and create a signal instead. 

### Changes

* Adds CreateAlert to schemas for validation
* Hides all dynamic alert funcs from test output except for the alert context

### Testing

* All current tests pass
* Manually tested test and upload with the new field

Example output of upload cmd:
```bash
/Users/mcr/.local/share/virtualenvs/max-pa-0qsMSnqL/bin/pat:4: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  __import__('pkg_resources').require('panther-analysis-tool==0.36.0')
[INFO][root]: Found Environment Variables prefixed with 'PANTHER'.
[INFO][root]: Panther Public API endpoint: https://rmbm7y7318.execute-api.us-west-2.amazonaws.com/v1/public/graphql
[INFO][root]: Testing analysis items in signals/

[WARNING][root]: Skipping Correlation Rule 'bulk.upload.cr.with.signal.create.alert.false', testing not supported
bulk.upload.rule.create.alert.true.derived
	[PASS] pass
		[PASS] [rule] true
		[PASS] [dedup] defaultDedupString:bulk.upload.rule.create.alert.true.derived
	[PASS] fail
		[PASS] [rule] false

bulk.upload.rule.create.alert.false
	[PASS] pass
		[PASS] [rule] true
	[PASS] fail
		[PASS] [rule] false

bulk.upload.rule.create.alert.missing
	[PASS] action is list detections
		[PASS] [rule] true
		[PASS] [dedup] defaultDedupString:bulk.upload.rule.create.alert.missing
	[PASS] action is list bananas
		[PASS] [rule] false

bulk.upload.scheduled.rule.create.alert.false
	[PASS] pass
		[PASS] [rule] true
	[PASS] fail
		[PASS] [rule] false

--------------------------
Panther CLI Test Summary
	Path: signals/
	Passed: 5
	Failed: 0
	Invalid: 0

[INFO][root]: Zipping analysis items in signals/ to .
[INFO][root]: Uploading items to Panther
[INFO][root]: API Response:
{
    "rules": {
        "new": 2,
        "total": 4,
        "modified": 0
    },
    "queries": {
        "new": 0,
        "total": 0,
        "modified": 0
    },
    "policies": {
        "new": 0,
        "total": 0,
        "modified": 0
    },
    "data_models": {
        "new": 0,
        "total": 28,
        "modified": 0
    },
    "lookup_tables": {
        "new": 0,
        "total": 0,
        "modified": 0
    },
    "global_helpers": {
        "new": 0,
        "total": 34,
        "modified": 0
    },
    "correlation_rules": {
        "new": 0,
        "total": 0,
        "modified": 0
    }
}
[INFO][root]: Upload succeeded
```
